[BUG] The plugin cannot access the meta data when running behind a proxy

**Describe the bug**
The plugin cannot access the meta data from `ossindex.sonatype.org` when running behind a proxy without direct internet access.

**To Reproduce**
Steps to reproduce the behavior:
1. Configure a gradle build with the plugin
2. Configure a proxy in `gradle.properties` (`systemProp.https.proxyHost`, `systemProp.https.proxyPort`)
3. Run `./gradlew ossIndexAudit`
4. See error `Connection timed out`

**Expected behavior**
The Java system properties for Proxy settings should be applied so the meta data can be fetched from behind a proxy.

**Screenshots**
```
> Task :ossIndexAudit
Using anonymous request
Checking vulnerabilities in 120 dependencies

> Task :ossIndexAudit FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':ossIndexAudit'.
> Could not audit the project: Connect to ossindex.sonatype.org:443 [ossindex.sonatype.org/3.128.15.143, ossindex.sonatype.org/18.218.3.80, ossindex.sonatype.org/3.133.173.34] failed: Operation timed out (Connection timed out)
```

**Desktop (please complete the following information):**
 - OS: MacOS 10.15.5
 - Gradle Version: 6.4.1
 - JVM Version and Flavor: Open JDK 8
 - Version: 1.2.0

--- a/README.md
+++ b/README.md
@@ -69,6 +69,13 @@ ossIndexAudit {
     cacheExpiration = 'PT12H' // 12 hours if omitted. It must follow the Joda Time specification at https://www.javadoc.io/doc/joda-time/joda-time/2.10.4/org/joda/time/Duration.html#parse-java.lang.String-
     colorEnabled = false // if true prints vulnerability description in color. By default is true.
     dependencyGraph = true // if true prints dependency graph showing direct/transitive dependencies. By default is false.  
+    proxyConfiguration { // extra configuration when running behind a proxy without direct internet access
+        protocol = 'http' // can be 'http' (default) or 'https'
+        host = 'proxy-host' // hostname for the proxy
+        port = 8080 // port for the proxy
+        authConfiguration.username = 'username' // username for the proxy (if credentials are required)
+        authConfiguration.password = 'password' // password for the proxy (if credentials are required)
+    }
 }
 ```
 - Open Terminal on the project's root and run `./gradlew ossIndexAudit`

--- a/src/main/java/org/sonatype/gradle/plugins/scan/ossindex/OssIndexClientConfigurationBuilder.java
+++ b/src/main/java/org/sonatype/gradle/plugins/scan/ossindex/OssIndexClientConfigurationBuilder.java
@@ -65,6 +65,8 @@ public class OssIndexClientConfigurationBuilder
 
         clientConfiguration.setCacheConfiguration(cacheConfig);
       }
+
+      clientConfiguration.setProxyConfiguration(extension.getProxyConfiguration());
     }
 
     return clientConfiguration;

--- a/src/main/java/org/sonatype/gradle/plugins/scan/ossindex/OssIndexPluginExtension.java
+++ b/src/main/java/org/sonatype/gradle/plugins/scan/ossindex/OssIndexPluginExtension.java
@@ -15,6 +15,11 @@
  */
 package org.sonatype.gradle.plugins.scan.ossindex;
 
+import org.sonatype.ossindex.service.client.transport.AuthConfiguration;
+import org.sonatype.ossindex.service.client.transport.ProxyConfiguration;
+
+import groovy.lang.Closure;
+import org.apache.commons.lang3.StringUtils;
 import org.gradle.api.Project;
 
 public class OssIndexPluginExtension
@@ -43,6 +48,8 @@ public class OssIndexPluginExtension
   private boolean colorEnabled;
 
   private boolean dependencyGraph;
+
+  private ProxyConfiguration proxyConfiguration;
 
   public OssIndexPluginExtension(Project project) {
     username = "";
@@ -134,5 +141,23 @@ public class OssIndexPluginExtension
 
   public void setDependencyGraph(boolean dependencyGraph) {
     this.dependencyGraph = dependencyGraph;
+  }
+
+  public ProxyConfiguration getProxyConfiguration() {
+    return proxyConfiguration;
+  }
+
+  public void setProxyConfiguration(Closure<ProxyConfiguration> closure) {
+    proxyConfiguration = new ProxyConfiguration();
+    AuthConfiguration authConfiguration = new AuthConfiguration();
+    proxyConfiguration.setAuthConfiguration(authConfiguration);
+
+    closure.setResolveStrategy(Closure.DELEGATE_FIRST);
+    closure.setDelegate(proxyConfiguration);
+    closure.call();
+
+    if (StringUtils.isAllBlank(authConfiguration.getUsername(), authConfiguration.getPassword())) {
+      proxyConfiguration.setAuthConfiguration(null);
+    }
   }
 }

--- a/src/test/java/org/sonatype/gradle/plugins/scan/ossindex/OssIndexClientConfigurationBuilderTest.java
+++ b/src/test/java/org/sonatype/gradle/plugins/scan/ossindex/OssIndexClientConfigurationBuilderTest.java
@@ -20,7 +20,10 @@ import java.nio.file.Paths;
 import org.sonatype.ossindex.service.client.OssindexClientConfiguration;
 import org.sonatype.ossindex.service.client.cache.DirectoryCache;
 import org.sonatype.ossindex.service.client.cache.DirectoryCache.Configuration;
+import org.sonatype.ossindex.service.client.transport.AuthConfiguration;
+import org.sonatype.ossindex.service.client.transport.ProxyConfiguration;
 
+import groovy.lang.Closure;
 import org.joda.time.Duration;
 import org.junit.Before;
 import org.junit.Test;
@@ -53,6 +56,7 @@ public class OssIndexClientConfigurationBuilderTest
     assertThat(result).isNotNull();
     assertThat(result).isEqualToIgnoringGivenFields(expected, "cacheConfiguration");
     assertThat(result.getCacheConfiguration()).isInstanceOf(DirectoryCache.Configuration.class);
+    assertThat(result.getProxyConfiguration()).isNull();
   }
 
   @Test
@@ -100,5 +104,72 @@ public class OssIndexClientConfigurationBuilderTest
 
     Configuration cacheConfiguration = (Configuration) result.getCacheConfiguration();
     assertThat(cacheConfiguration.getExpireAfter()).isEqualTo(expectedDuration);
+  }
+
+  @Test
+  @SuppressWarnings("serial")
+  public void testBuild_usingProxy() {
+    OssIndexPluginExtension extension = new OssIndexPluginExtension(null);
+    extension.setProxyConfiguration(new Closure<ProxyConfiguration>(null)
+    {
+      @Override
+      public ProxyConfiguration call() {
+        ProxyConfiguration proxyConfiguration = (ProxyConfiguration) getDelegate();
+        proxyConfiguration.setHost("test-host");
+        proxyConfiguration.setPort(8080);
+        return proxyConfiguration;
+      }
+    });
+
+    OssindexClientConfiguration result = builder.build(extension);
+
+    assertThat(result.getProxyConfiguration()).isNotNull();
+    assertThat(result.getProxyConfiguration().getHost()).isEqualTo("test-host");
+    assertThat(result.getProxyConfiguration().getPort()).isEqualTo(8080);
+    assertThat(result.getProxyConfiguration().getProtocol()).isEqualTo(ProxyConfiguration.HTTP);
+    assertThat(result.getProxyConfiguration().getAuthConfiguration()).isNull();
+  }
+
+  @Test
+  @SuppressWarnings("serial")
+  public void testBuild_usingProxyHttps() {
+    OssIndexPluginExtension extension = new OssIndexPluginExtension(null);
+    extension.setProxyConfiguration(new Closure<ProxyConfiguration>(null)
+    {
+      @Override
+      public ProxyConfiguration call() {
+        ProxyConfiguration proxyConfiguration = (ProxyConfiguration) getDelegate();
+        proxyConfiguration.setProtocol(ProxyConfiguration.HTTPS);
+        return proxyConfiguration;
+      }
+    });
+
+    OssindexClientConfiguration result = builder.build(extension);
+
+    assertThat(result.getProxyConfiguration().getProtocol()).isEqualTo(ProxyConfiguration.HTTPS);
+  }
+
+  @Test
+  @SuppressWarnings("serial")
+  public void testBuild_usingProxyWithAuthentication() {
+    OssIndexPluginExtension extension = new OssIndexPluginExtension(null);
+    extension.setProxyConfiguration(new Closure<ProxyConfiguration>(null)
+    {
+      @Override
+      public ProxyConfiguration call() {
+        ProxyConfiguration proxyConfiguration = (ProxyConfiguration) getDelegate();
+        AuthConfiguration authConfiguration = proxyConfiguration.getAuthConfiguration();
+        authConfiguration.setUsername("test-username");
+        authConfiguration.setPassword("test-password");
+        return proxyConfiguration;
+      }
+    });
+
+    OssindexClientConfiguration result = builder.build(extension);
+
+    AuthConfiguration authConfiguration = result.getProxyConfiguration().getAuthConfiguration();
+    assertThat(authConfiguration).isNotNull();
+    assertThat(authConfiguration.getUsername()).isEqualTo("test-username");
+    assertThat(authConfiguration.getPassword()).isEqualTo("test-password");
   }
 }

